src/components/Modals/EditExpertProfileModal.tsx

```diff
@@ -0,0 +1,166 @@
+      email == null ||
+      expertise == null ||
+      description == null ||
+      expertiseTags == null ||
+      visibility == null
+    ) {
+      setFieldsMissing(true);
+      return;
+    }
+    apiContext
+      .updateJufoExpert(user.id, {
+        contactEmail: email,
+        description,
+        expertiseTags: expertiseTags.split(','),
+        active: visibility,
+      })
+      .then(() => closeModal());
+  };
+
+  return (
+    <DialogModalBase accentColor={accentColor}>
+      <DialogModalBase.Modal modalName="editExpertProfileModal">
+        <DialogModalBase.Header>
+          <div className={styles.avatarWrapper}>
+            <Avatar className={styles.avatar} />
+            <button
+              className={styles.avatarOverlay}
+              onClick={() => avatarUploader.current.click()}
+            >
+              <p>Ändern</p>
+              <input
+                type="file"
+                id="file"
+                ref={avatarUploader}
+                accept="image/*"
+                style={{ display: 'none' }}
+                onChange={(e) => {
+                  e.stopPropagation();
+                  e.preventDefault();
+                  const file = e.target.files[0];
+                  console.log(file);
+                  const fileReader = new FileReader();
+                  fileReader.onload = () => {
+                    setAvatar(() => ({ className }) => {
+                      return (
+                        <img
+                          className={className}
+                          alt="avatar"
+                          src={String(fileReader.result)}
+                        />
+                      );
+                    });
+                  };
+
+                  fileReader.readAsDataURL(file);
+                }}
+              />
+            </button>
+          </div>
+        </DialogModalBase.Header>
+        <DialogModalBase.Spacer />
+        <DialogModalBase.Header>
+          <DialogModalBase.Title>Melanie Meiers</DialogModalBase.Title>
+          <DialogModalBase.CloseButton />
+        </DialogModalBase.Header>
+        {fieldsMissing && (
+          <DialogModalBase.Error>
+            <span>Bitte alle Felder ausfüllen.</span>
+          </DialogModalBase.Error>
+        )}
+        <DialogModalBase.Form>
+          <DialogModalBase.InputCompound direction="vertical">
+            <DialogModalBase.TextBox
+              label="Kontakt E-Mail-Adresse"
+              onChange={(e) => setEmail(e.target.value)}
+              value={email}
+            />
+            <DialogModalBase.TextBox
+              label="Fachgebiet"
+              onChange={(e) => setExpertise(e.target.value)}
+              value={expertise}
+            />
+            <DialogModalBase.TextArea
+              label="Kurzbeschreibung"
+              onChange={(e) => setDescription(e.target.value)}
+              value={description}
+            />
+            <DialogModalBase.TextBox
```

Comment:
It would be nice if the user would get some proposals as available through `ApiContext.getUsedExpertTags`. Furthermore the expertiseTags must be a list of strings. Now only one tag is saved no matter what I type in. Mayebe use something like [this](https://ant.design/components/select/) (with automatic tokenization) here.

Reply:
Closed in e03aa05